### PR TITLE
release-2024.6.1-hotfix:

### PR DESCRIPTION
**Pull request by f1sh1918**
### Short description

Gold cards can not be activated since the `expirationDate `  was wrongly set to `0`

### Proposed changes

- check if card has expirationDay and only set it then, else null
- add migration to fix cards that have `expirationDay = 0`
- add missing null check for startDay for staticCards

### Side effects

- none

### Resolved issues

Fixes: #1517

### Testing
- Create a gold card
- Try to activate the gold card
- card should not be expired anymore and can be activated

### CODE REVIEW

**Comment by michael-markl** [comment]
As I mentioned in the other pr, this is not the root of the cause.
ExpirationDay should be null for the golden card.

Please fix it in the card mutation service, and then add a migration script that sets those entries to null that have the value 0.

File: backend/src/main/kotlin/app/ehrenamtskarte/backend/verification/webservice/schema/CardMutationService.kt [hunk 1]
```diff
@@ -70,7 +70,7 @@ class CardMutationService {
             CardRepository.insert(
                 hashedCardInfo,
                 activationSecretHash,
-                cardInfo.expirationDay.toLong(),
+                expirationDay = if (cardInfo.hasExpirationDay()) cardInfo.expirationDay.toLong() else null,
```
Comment (michael-markl) [inline backend/src/main/kotlin/app/ehrenamtskarte/backend/verification/webservice/schema/CardMutationService.kt:73]: To make it consistent with the other optional fields:
```suggestion
                expirationDay = cardInfo.expirationDayOrNull?.toLong(),
```

Also, the same must be done for `createStaticVerificationCode`

Comment (f1sh1918) [reply backend/src/main/kotlin/app/ehrenamtskarte/backend/verification/webservice/schema/CardMutationService.kt:73]: updated pr

**Review by ztefanie — APPROVED** [review]
Thanks for fixing this. Tested on Android emulator, works as expected.

File: backend/src/main/kotlin/app/ehrenamtskarte/backend/verification/webservice/schema/CardMutationService.kt (hunk 1 above)
Comment (michael-markl) [reply backend/src/main/kotlin/app/ehrenamtskarte/backend/verification/webservice/schema/CardMutationService.kt:73]: @f1sh1918 I don't think so 🤔 at least I cannot see it yet.

Comment (f1sh1918) [reply backend/src/main/kotlin/app/ehrenamtskarte/backend/verification/webservice/schema/CardMutationService.kt:73]: sorry ssh timeout :/ gonna push it soon

Comment (f1sh1918) [reply backend/src/main/kotlin/app/ehrenamtskarte/backend/verification/webservice/schema/CardMutationService.kt:73]: @michael-markl now its pushed

**Review by michael-markl — APPROVED** [review]
If you have double checked that the DB migration does not update any (other) existing cards, this looks good to me :)

**Pull request merged**